plugins/services/src/js/components/modals/NewCreateServiceModal.js

```diff
@@ -1,51 +1,166 @@
-      serviceFormActive: false
-    };
+    this.state = this.getResetState();
 
     METHODS_TO_BIND.forEach((method) => {
       this[method] = this[method].bind(this);
     });
   }
 
+  componentWillUpdate(nextProps) {
+    const requestCompleted = this.props.isPending && !nextProps.isPending;
+    const shouldClose = requestCompleted && !nextProps.errors;
+
+    if (shouldClose) {
+      this.handleClose();
+    }
+  }
+
+  handleGoBack() {
+    let {
+      serviceFormActive,
+      servicePickerActive,
+      serviceReviewConfig
+    } = this.state;
+
+    // Close if picker is open, or if editing a service in the form
+    if (servicePickerActive || (serviceFormActive && this.props.isEdit)) {
+      this.handleClose();
+      return;
+    }
+
+    if (serviceFormActive) {
+      this.setState({
+        servicePickerActive: true,
+        serviceFormActive: false,
+        serviceReviewConfig: null
+      });
+      return;
+    }
+
+    if (serviceReviewConfig) {
+      this.setState({
+        servicePickerActive: false,
+        serviceFormActive: true,
+        serviceReviewConfig: null
+      });
+    }
+  }
+
+  handleClose() {
+    this.props.onClose();
+    this.setState(this.getResetState());
+  }
+
   handleJSONToggle() {
     this.setState({isJSONModeActive: !this.state.isJSONModeActive});
   }
 
+  getResetState(nextProps = this.props) {
+    let newState = {
+      isJSONModeActive: false,
+      serviceFormActive: false,
+      servicePickerActive: true,
+      serviceReviewConfig: null
+    };
+
+    // Switch directly to form if edit
+    if (nextProps.isEdit) {
+      newState.servicePickerActive = false;
+      newState.serviceFormActive = true;
+    }
+
+    return newState;
+  }
+
   handleServiceSelection() {
     this.setState({
       servicePickerActive: false,
-      serviceFormActive: true
+      serviceFormActive: true,
+      serviceReviewConfig: null
     });
   }
 
+  handleServiceReview() {
+    this.setState({
+      servicePickerActive: false,
+      serviceFormActive: false,
+      serviceReviewConfig: this.serviceModalForm.getAppConfig()
+    });
+  }
+
+  handleServiceRun() {
+    let {marathonAction, service} = this.props;
+    marathonAction(
+      service,
+      this.state.serviceReviewConfig
+    );
+  }
+
   getHeader() {
+    if (this.state.serviceReviewConfig) {
+      return (
+        <FullScreenModalHeader>
+          <FullScreenModalHeaderActions
+            actions={this.getSecondaryActions()}
+            type="secondary" />
+          <FullScreenModalHeaderTitle>
+            Review & Run Service
+          </FullScreenModalHeaderTitle>
+          <FullScreenModalHeaderActions
+            actions={this.getPrimaryActions()}
+            type="primary" />
+        </FullScreenModalHeader>
+      );
+    }
+
+    let title = 'Run a Service';
+    let {isEdit, service} = this.props;
+    let serviceName = service.getName() || 'Service';
+    if (serviceName) {
```

Comment:
Removed the `|| 'Service'` as that was a mistake. Had a conversation offline and I decided to keep the `serviceName` outside of the `if (isEdit)` to keep it easier to parse.